src/osgEarth/MapNode.cpp

```diff
@@ -388,14 +388,22 @@ MapNode::init()
         OE_INFO << LC << "Adding ellipsoid uniforms.\n";
 
         // for a geocentric map, use an ellipsoid unit-frame transform and its inverse:
-        osg::Vec3d ellipFrameInverse(
-            _map->getSRS()->getEllipsoid()->getRadiusEquator(),
-            _map->getSRS()->getEllipsoid()->getRadiusEquator(),
-            _map->getSRS()->getEllipsoid()->getRadiusPolar());
-        stateset->addUniform( new osg::Uniform("oe_ellipsoidFrameInverse", osg::Vec3f(ellipFrameInverse)) );
-
-        osg::Vec3d ellipFrame = osg::componentDivide(osg::Vec3d(1.0,1.0,1.0), ellipFrameInverse);
-        stateset->addUniform( new osg::Uniform("oe_ellipsoidFrame", osg::Vec3f(ellipFrame)) );
+        if (_map->getSRS() != NULL && _map->getSRS()->getEllipsoid() != NULL)
+        {
+            osg::Vec3d ellipFrameInverse(
+                _map->getSRS()->getEllipsoid()->getRadiusEquator(),
+                _map->getSRS()->getEllipsoid()->getRadiusEquator(),
+                _map->getSRS()->getEllipsoid()->getRadiusPolar());
+            stateset->addUniform( new osg::Uniform("oe_ellipsoidFrameInverse", osg::Vec3f(ellipFrameInverse)) );
+
+            osg::Vec3d ellipFrame = osg::componentDivide(osg::Vec3d(1.0,1.0,1.0), ellipFrameInverse);
+            stateset->addUniform( new osg::Uniform("oe_ellipsoidFrame", osg::Vec3f(ellipFrame)) );
+        }
+        else
+        {
+            stateset->addUniform( new osg::Uniform("oe_ellipsoidFrameInverse", osg::Vec3f()) );
```

Comment:
I am unsure if this is the best default value for oe_ellipsoidFrameInverse (and oe_ellipsoidFrame).  Would osg::Vec3f(1.f, 1.f, 1.f) be better?